crates/orderbook/src/orderbook.rs

```diff
@@ -277,4 +355,72 @@ mod tests {
             .unwrap();
         assert_eq!(result, &orders[1..2]);
     }
+
+    #[test]
+    fn computes_u256_prices_normalized_to_1e18() {
+        assert_eq!(
+            to_normalized_price(0.5),
+            U256::from(500_000_000_000_000_000_u128)
+        );
+    }
+
+    #[tokio::test]
+    async fn filters_tokens_without_native_prices() {
+        let token1 = H160([1; 20]);
+        let token2 = H160([2; 20]);
+        let token3 = H160([3; 20]);
+
+        let orders = vec![
```

Comment:
nit: I think covering all possibilities would be nice. That would need a test where neither `sell_token` nor `buy_token` of an order has a native price.

Reply:
Good suggestion! Added.